### ISSUE TITLE
Plenary: Open Science Discussion

### ISSUE THREAD

**Issue by scottyhq** [uwhackweek/schedule-2024#29]
**Lead:** Andy Barrett
**Date:** 23/08/2024
**Start Time:** 0900
**Duration:** 45
**Description:** Open science best practices

<details><summary>Details</summary>
<p>

### Learning Outcomes

* outcome1
* outcome2
* outcome3

### Dependencies (things people should know in advance of the tutorial)

Not dependencies, just adding a few relevant links to previous content

* https://icesat-2-2023.hackweek.io/reference/open_science.html
* https://book.cryointhecloud.com/content/about.html
* https://www.earthdata.nasa.gov/technology/open-science

### Technical Needs (GPUs? Large file storage? Unique libraries?)

</p>
</details>

**Comment by andypbarrett** [comment]
Happy to lead development of this session.

Does it make sense to have this as a panel discussion, or quick pop-ups, and maybe a Q&A.

**Comment by JessicaS11** [comment]
> Does it make sense to have this as a panel discussion, or quick pop-ups, and maybe a Q&A.

Those are all fun ideas!

**Comment by andypbarrett** [comment]
Here's another thought I had.

My initial idea was to have a discussion on producing reusable and interpretable data.  I was thinking about providing groups with various data with missing attributes etc and asking them to recreate a plot.

The best-case scenario would be a CF-Compliant NetCDF.  The worst-case, a flat binary.  This would hopefully seed a discussion on how to create data that can be reused out-of-the box.

**Comment by JessicaS11** [comment]
> My initial idea was to have a discussion on producing reusable and interpretable data. I was thinking about providing groups with various data with missing attributes etc and asking them to recreate a plot.

I love this idea! It's such a good way to illustrate why this is so important.

**Comment by tsutterley** [comment]
That's a super cool idea 💡

**Comment by dphow** [comment]
As a reference to add, at NOAA/NCAR/NREL's hackathon earlier this year, we hosted a speaker from AGU Open Science Leadership. They gave a great talk to our participants about Open Science. Feel free to use this as a resource to inspire this session or panel discussion questions: https://zenodo.org/records/10720248

**Comment by aaarendt** [comment]
Adding @mstudinger to this thread who I've been speaking with regarding his role as a NASA TOPS Champion. Michael has been teaching the Open Science 101 curriculum to summer interns and offered to join in this session.

Suggest all of us hop on a call sometime soon?

**Comment by andypbarrett** [comment]
Apologies for not getting back to this sooner.  I've been moving house from Denver to Taos, NM.

@aaarendt and @mstudinger Happy to jump on a call Thursday 8 August (tomorrow) or Friday 9 August.  Could also do early next week.

**Comment by mstudinger** [comment]
Happy to talk/brainstorm about this session. I really like the idea of asking participants to create a reusable and interpretable data set or science result. I am not sure if we can do that in 45 minutes though, but let’s definitely think about it.

Here are few random thoughts (mostly based on my experience teaching OS/advocating for it in the community)
I think the title “open science best practices” is a great focus of what this session could be. Given the focus of the hackweeks on open science, open collaboration, etc I don’t think we need to convince anyone at the hackweek that open science is a good thing. But in my experience people have still a lot of anxieties with something new, just like there was when NASA switched to the Dual-Anonymous Peer Review (DAPR) a few years ago. When I talk about open science folks are typically very keen to ask questions and share their concerns. We should allocate a good amount of time for questions/discussions if possible.

The monstrous 12.5 hour NASA curriculum covers mostly **how to use, make and share open data, open code, and open results** and could be a structure for this session as well. Things like: when it is ok to be open and when it is not ok to be open? Why should you give your data/code/result a license? After taking the training earlier this year I assigned licenses to all my git repos….  I hadn’t bothered doing that before because I didn’t know…. How do you search for open data/code/results and evaluate the quality? How do you make you data/code/result findable (Zenodo DOI vs git repo/web address). Things like that.
 
As already mentioned by others there is a lot of material available on the web. Not that it would be needed, but if people are interested here is the NASA OS 101 curriculum on github: 
https://github.com/nasa/Transform-to-Open-Science

The 5 modules with 5 lessons each are here: https://github.com/nasa/Transform-to-Open-Science/tree/open-science-101

**Comment by andypbarrett** [comment]
I agree 45 minutes is not much time and also that we do not need to duplicate the TOPS curriculum, or evangelize open science.  Hopefully, we would be preaching to the choir.
meeting of how 
Maybe we should title the session "Putting open science into practice" and try to be practical rather than prescriptive.

I think it is also worth emphasizing that the open science best practices not only benefit the community and satisfy funding agency or publisher requirements, but also benefit our "future selves" when we come back to a project.  I've been involved with Openscapes for a while.  I am very drawn to emphasizing that open science is about improving collaboration (Anthony showed a great graphic of how improving interactions can be transformative in yesterdays meeting) within research groups, between groups and for the community in general.  None of this can happen without openness and more importantly without creating psychologically safe spaces where everyone can be vulnerable and adopt a learning mindset.

I think topics that would be of benefit may fall along what are the next steps for projects.  So things like

- Using Github and Googledocs to document work and discussion.  
- Creating dynamic lab guidebooks - the books from the hackweeks are a great example of this. 
- Making data interoperable and reusable
    - data formats
    - conventions
    - metadata  
    - cloud optimized formats - I think Aimee will cover some of that
- Licenses
- PIDS
- Archives

**Comment by andypbarrett** [comment]
@mstudinger just pushed to https://github.com/andypbarrett/website-2024/tree/open-science-plenary and created draft PR https://github.com/ICESAT-2HackWeek/website-2024/pull/31
